### PR TITLE
[RND-21] API Parity and Bulk Load Support

### PR DESCRIPTION
**Pull request by stephenfuqua**

### CODE REVIEW

File: Meadowlark-js/services/meadowlark-fastify/src/Service.ts [hunk 1]
```diff
@@ -3,51 +3,128 @@
...
+    // Swagger handlers
+    fastify.get(`/${stage}/metadata`, openApiUrlList);
+    fastify.get(`/${stage}/metadata/`, openApiUrlList);
+    fastify.get(`/${stage}/metadata/resources/swagger.json`, swaggerForResourcesAPI);
+    fastify.get(`/${stage}/metadata/descriptors/swagger.json`, swaggerForDescriptorsAPI);
+    fastify.get(`/${stage}/metadata/data/v3/dependencies`, dependencies);
 
-  return service;
+    // OAuth handlers
+    fastify.post(`/${stage}/api/oauth/token`, oauthHandler);
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply Meadowlark-js/services/meadowlark-fastify/src/Service.ts:126]: False alarm - the rate limiting _is_ configured, but CodeQL isn't seeing it.

File: Meadowlark-js/tests/system/crud/SystemTestSetup.ts [hunk 2]
```diff
@@ -89,3 +93,9 @@ export function academicWeekBodyClient1(): FrontendRequest {
     }`,
   };
 }
+
+export function configureEnvironmentForMongoSystemTests(): void {
+  process.env.DOCUMENT_STORE_PLUGIN = MONGO_DOCUMENT_STORE_PLUGIN;
+  process.env.SIGNING_KEY = TEST_SIGNING_KEY;
+  process.env.MEADOWLARK_DATABASE_NAME = 'meadowlark_system_tests';
+}
```
Comment (bradbanister) [inline Meadowlark-js/tests/system/crud/SystemTestSetup.ts:101]: One thought I had was for DOCUMENT_STORE_PLUGIN to be set external to the tests. That way they have no dependency on the particular document store, so the system tests for the various document stores could be run as npm scripts that just differ in the enviroment variables, e.g. `cross-env DOCUMENT_STORE_PLUGIN=@edfi/meadowlark-mongodb-backend yarn jest system --runInBand`

File: Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Db.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ import { Logger } from '@edfi//meadowlark-core';
 import { MeadowlarkDocument } from '../model/MeadowlarkDocument';
 
 const MONGO_URL_DEFAULT = 'mongodb://mongo1:27017,mongo2:27018,mongo3:27019/?replicaSet=rs0';
-export const DATABASE_NAME = 'meadowlark';
+export const DATABASE_NAME = process.env.MEADOWLARK_DATABASE_NAME || 'meadowlark';
```
Comment (bradbanister) [inline Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Db.ts:11]: Something to keep in mind is that more configurability means more testing is needed. For example, the Debezium connector uses the database name as part of the Kafka topic name.

Comment (stephenfuqua) [reply Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Db.ts:11]: I do think its important to let this vary. In particular I was motivated by not wanting to destroy my database every time I run system or integration tests.

File: Meadowlark-js/tests/system/crud/SystemTestSetup.ts (hunk 2 above)
Comment (stephenfuqua) [reply Meadowlark-js/tests/system/crud/SystemTestSetup.ts:101]: We'll definitely want to refactor in that direction.

**Pull request merged**